source/engine/graphics/system/SkeletalAnimationSystem.cpp

```diff
@@ -167,24 +149,23 @@ void SkeletalAnimationSystem::UpdateBonesStorage(const asset::BoneUpdateEventDat
 
 void SkeletalAnimationSystem::Start(const anim::PlayState& playState)
 {
-    if (m_rigs.find(playState.meshUid) == m_rigs.end() ||
-        m_animationAssets.find(playState.curAnimUid) == m_animationAssets.end() ||
-        m_animationAssets.find(playState.prevAnimUid) == m_animationAssets.end())
+    if (!m_rigs.contains(playState.meshUid) ||
+        (!m_animationAssets.contains(playState.curAnimUid) &&
+        !m_animationAssets.contains(playState.prevAnimUid)) )
     {
         return;
     }
 
-    auto pos = std::find_if(m_unitEntities.begin(), m_unitEntities.end(), [playState](Entity entity){ return playState.entity.Index() == entity.Index(); });
-
+    auto pos = std::ranges::find(m_unitEntities, playState.entity);
     if (pos == m_unitEntities.end())
     {
-        m_unitEntities.push_back(playState.entity);
-        m_units.push_back(anim::UnitOfWork
+        m_unitEntities.emplace_back(playState.entity);
+        m_units.emplace_back(anim::UnitOfWork
```

Comment:
`emplace_back()` is used for in-place construction by forwarding arguments to the constructor. If you construct a `UnitOfWork` at the call site here, the in-place construction happens via a move constructor (aka two constructors run instead of one):
```cpp
m_units.emplace_back
(
    playState.entity.Index(),
    &m_rigs.at(playState.meshUid),
    playState.prevAnimUid.empty() ? nullptr : &m_animationAssets.at(playState.prevAnimUid),
    playState.curAnimUid.empty() ? nullptr : &m_animationAssets.at(playState.curAnimUid),
    0.0f,
    0.0f,
    0.0f
);
```